bayesian/api_v1.py

```diff
@@ -583,6 +585,32 @@ def get_component_analysis_schema_url(cls, name, version):
                                    name=name, version=version)
 
 
+class GenerateManifest(Resource):
+    method_decorators = [login_required]
+
+    @staticmethod
+    def post():
+        input_json = request.get_json()
+        if 'ecosystem' not in input_json:
+            raise HTTPError(400, "Must provide a ecosystem")
+        if input_json.get('ecosystem') == 'maven':
+            return Response(
+                PomXMLTemplate(input_json).xml_file(),
+                headers={
+                    "Content-disposition": 'attachment;filename=pom.xml',
+                    "Content-Type": "text/xml;charset=utf-8"
+                }
+            )
+        elif input_json.get('ecosystem') == 'node':
```

Comment:
What about having directly else branch so if we add a new ecosystem, the error message would be reusable (in other words do not hardcode ecosystem names in sources if not necessary)?
```python
else:
    return {'result': "ecosystem '{}' is not yet supported".format(input_json['ecosystem'])}
```

Also return 400 HTTP status code to signalize error instead of default 200.

Reply:
yes, this makes sense. I will change this.